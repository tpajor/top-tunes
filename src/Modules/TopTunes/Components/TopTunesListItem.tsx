import * as React from 'react';
import { Link } from 'react-router-dom';

import { urls } from '../../../Routing/urls';
import IITune from '../../../Shared/Interfaces/models/IITune';

export interface IProps {
  item: IITune;
  position: number;
  iteration: number;
}

const TopTunesListItem = ({ item, position, iteration }: IProps) => {
  return (
    <Link
      className={`link link--list-item${iteration % 2 === 0 ? '-even' : '-odd'}`}
      to={urls.reactRouter.topTunesDetailCreate(position)}
    >
      <h5 className='text text--list-item-title'>
        <span className='text text--list-item-title text--highlight'>
          {position + 1}
        </span>
        {item.title}
      </h5>
    </Link>
  );
};

export default TopTunesListItem;
