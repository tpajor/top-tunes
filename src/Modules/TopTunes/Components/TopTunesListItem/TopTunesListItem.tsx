import * as React from 'react';

interface IProps {
  item: any;
  position: number;
}

const TopTunesListItem = ({ item, position }: IProps) => {
  return (
    <div className='container container--list-item'>
      <h5 className='text text--list-item-title'>
        <span className='text text--list-item-title text--highlight'>
          {+position + 1}
        </span>
        {item.title.label}
      </h5>
    </div>
  );
};

export default TopTunesListItem;
