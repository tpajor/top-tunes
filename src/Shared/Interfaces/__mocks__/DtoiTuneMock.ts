import IDtoiTune from '../dto/IDtoiTune';

export const dtoiTuneMock: IDtoiTune = {
  category: {
    attributes: {
      'im:id': 'mock',
      label: 'category',
      scheme: 'category schema',
      term: 'category term',
    },
  },

  id: {
    label: 'id',
    attributes: {
      'im:id': 'mock',
    },
  },

  'im:artist': {
    label: 'artist',
    attributes: {
      href: 'http://artist.com',
    },
  },

  'im:contentType': {
    attributes: {
      term: 'constentType term',
      label: 'constentType',
    },
    'im:contentType': {
      attributes: {
        term: 'constentType term',
        label: 'constentType',
      },
    },
  },

  'im:image': [
    {
      label: 'http://image.com',
      attributes: {
        height: '13',
      },
    },
  ],

  'im:itemCount': {
    label: '13',
  },

  'im:name': {
    label: 'name',
  },

  'im:price': {
    label: '$13.13',
    attributes: {
      amount: '13.13',
      currency: 'dollar',
    },
  },

  'im:releaseDate': {
    label: '1999-09-09',
    attributes: {
      label: '1999-09-09',
    },
  },

  link: {
    attributes: {
      href: 'http://link.com',
      rel: 'link',
      type: 'link type',
    },
  },

  rights: {
    label: 'rights',
  },

  title: {
    label: 'title',
  },
};
