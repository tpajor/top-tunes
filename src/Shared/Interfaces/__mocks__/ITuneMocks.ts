import IITune from '../models/IITune';

const iTune1: IITune = {
  category: 'Jazz',
  artist: 'Various',
  photo: 'http://photo-url.com',
  numberOfSongs: 14,
  name: 'Some Album',
  price: '$6.67',
  releaseDate: new Date(2, 2, 2, 2, 2, 2, 2),
  link: 'http://profile-url.com',
  rights: 'copyleft',
  title: 'Various - Some Album',
};

const iTune2: IITune = {
  category: 'Rock',
  artist: 'Artist Name',
  photo: 'http://photo-url-1.com',
  numberOfSongs: 8,
  name: 'Some Other Album',
  price: '$13.13',
  releaseDate: new Date(4, 4, 4, 4, 4, 4, 4),
  link: 'http://profile-url-1.com',
  rights: 'copyleft rights',
  title: 'Artist Name - Some Other Album',
};

const iTune3: IITune = {
  category: 'Pop',
  artist: 'Lazy Artist Name',
  photo: 'http://photo-url-2.com',
  numberOfSongs: 11,
  name: 'Yet Another Album',
  price: '$1.09',
  releaseDate: new Date(8, 8, 8, 8, 8, 8, 8),
  link: 'http://profile-url-2.com',
  rights: 'copyleft rights reversed',
  title: 'Lazy Artist Name - Yet Another Album',
};

export const fromDto: IITune = {
  category: 'category term',
  artist: 'artist',
  photo: 'http://image.com',
  numberOfSongs: 13,
  name: 'name',
  price: '$13.13',
  releaseDate: new Date('1999-09-09'),
  link: 'http://link.com',
  rights: 'rights',
  title: 'title',
};

export const iTunes: IITune[] = [
  iTune1,
  iTune2,
  iTune3,
];
