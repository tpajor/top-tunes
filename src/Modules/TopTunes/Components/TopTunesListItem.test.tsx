import * as React from 'react';
import { shallow } from 'enzyme';

import TopTunesListItem, { IProps } from './TopTunesListItem';
import { iTunes } from '../../../Shared/Interfaces/__mocks__/ITuneMocks';

function setup() {
  const props: IProps = {
    item: iTunes[0],
    position: 2,
    iteration: 2,
  };

  const wrapper = shallow(<TopTunesListItem {...props} />);

  return {
    wrapper,
    props,
  };
}

describe('TopTunesListItem component', () => {
  it('should match snapshot if no change was intended', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render link with even class if iteretion props is an even number', () => {
    const { wrapper, props } = setup();

    expect(wrapper.find('Link').hasClass('link link--list-item-even')).toBe(true);

    wrapper.setProps({ ...props, iteration: 1 });

    expect(wrapper.find('Link').hasClass('link link--list-item-odd')).toBe(true);
  });
});
