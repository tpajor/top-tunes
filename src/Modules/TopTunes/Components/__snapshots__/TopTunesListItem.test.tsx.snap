// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopTunesListItem component should match snapshot if no change was intended 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TopTunesListItem
    item={
      Object {
        "artist": "Various",
        "category": "Jazz",
        "link": "http://profile-url.com",
        "name": "Some Album",
        "numberOfSongs": 14,
        "photo": "http://photo-url.com",
        "price": "$6.67",
        "releaseDate": 1902-03-02T00:38:02.002Z,
        "rights": "copyleft",
        "title": "Various - Some Album",
      }
    }
    iteration={2}
    position={2}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <h5
        className="text text--list-item-title"
      >
        <span
          className="text text--list-item-title text--highlight"
        >
          3
        </span>
        Various - Some Album
      </h5>,
      "className": "link link--list-item-even",
      "replace": false,
      "to": "/topTunes/2",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="text text--list-item-title text--highlight"
          >
            3
          </span>,
          "Various - Some Album",
        ],
        "className": "text text--list-item-title",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 3,
            "className": "text text--list-item-title text--highlight",
          },
          "ref": null,
          "rendered": 3,
          "type": "span",
        },
        "Various - Some Album",
      ],
      "type": "h5",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <h5
          className="text text--list-item-title"
        >
          <span
            className="text text--list-item-title text--highlight"
          >
            3
          </span>
          Various - Some Album
        </h5>,
        "className": "link link--list-item-even",
        "replace": false,
        "to": "/topTunes/2",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="text text--list-item-title text--highlight"
            >
              3
            </span>,
            "Various - Some Album",
          ],
          "className": "text text--list-item-title",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 3,
              "className": "text text--list-item-title text--highlight",
            },
            "ref": null,
            "rendered": 3,
            "type": "span",
          },
          "Various - Some Album",
        ],
        "type": "h5",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
